refactor: batchKey logic

karatakis opened the pull request:
I rewrote the documentation because it was unclear how to use the API based on feedback from here https://github.com/tailcallhq/tailcall/pull/2495

karatakis committed:
Update documentation with the updated logic for batchKey and query field

karatakis committed:
Correct docs

karatakis committed:
Update docs

tusharmath commented on docs/directives.md:
```diff
@@ -882,7 +882,7 @@ type Mutation {
 ### query
 
-Represents the API call's query parameters, either as a static object or with dynamic parameters using Mustache templates. These parameters append to the URL.
+Represents the API call's query parameters, either as a static object or with dynamic parameters using Mustache templates. These parameters append to the URL. The `key` parameter of `query` is optional. If it is not present TailCall uses the last item from `batchKey` parameter of `@http` directive to derive the `key` parameter, so remember to adjust the order of the items accordingly.
```
I think this note should be written under an `:::important` section of `batchKey`

tusharmath replied:
Let's provide an actual example here for more clarity.

tusharmath commented:
@karatakis Can you please resolve the conflicts here.

tusharmath commented:
Moving to draft to reduce noise and improve CI efficiency. Once you are ready just mark it as "ready to review". Feel free to give a shoutout on the #contributors channel on Discord if you want immediate attention.

karatakis committed:
Merge branch 'develop' into update-batch-key-logic

karatakis committed:
Fix documentation

karatakis committed:
Improve documentation

karatakis committed:
Further improve the docs

karatakis commented:
@tusharmath I rewrote the docs to describe how Tailcall behaves when batchKey and query are present. I also changed the batchKey example to be more generic and describe better the edge case.

tusharmath commented:
Thanks!

tusharmath committed:
whitespace changes

tusharmath committed:
Merge branch 'develop' into update-batch-key-logic
